data_pipeline/elife_article_xml/elife_article_xml_pipeline.py

```diff
@@ -0,0 +1,124 @@
+import base64
+from typing import Iterable, Any, Optional, Tuple
+import logging
+import xml.etree.ElementTree as ET
+import requests
+
+
+from data_pipeline.elife_article_xml.elife_article_xml_config import (
+    ElifeArticleXmlConfig,
+    ElifeArticleXmlSourceConfig
+)
+from data_pipeline.utils.data_store.bq_data_service import (
+    get_single_column_value_list_from_bq_query,
+    load_given_json_list_data_from_tempdir_to_bq
+)
+from data_pipeline.utils.json import (
+    get_recursively_transformed_object,
+    remove_key_with_null_value
+)
+from data_pipeline.utils.xml import parse_xml_and_return_it_as_dict
+
+LOGGER = logging.getLogger(__name__)
+
+
+def get_json_response_from_url(url: str) -> Any:
+    response = requests.get(url=url)
+    response.raise_for_status()
+    return response.json()
+
+
+def get_url_of_xml_file_directory_from_repo(
+    source_config: ElifeArticleXmlSourceConfig,
+) -> str:
+    response_json = get_json_response_from_url(url=source_config.git_repo_url)
+    for folder in response_json['tree']:
+        if folder['path'] == source_config.directory_name:
+            return folder['url']
+
+
+def iter_xml_file_url_from_git_directory(
+    source_config: ElifeArticleXmlSourceConfig,
+    processed_file_url_list: Iterable[str]
```

Comment:
I think the type would need to be `Sequence` or `Set`. (`Iterable` wouldn't work because it can be iterated only once)

Reply:
Just a thought, skipping processed files could also be done in a separate function.

Reply:
changed the file name to `iter_unprocessed_xml_file_url_from_git_directory`

Reply:
(just leaves the type annotation)